### PR TITLE
feat: Add contract cancellation initiator to the API

### PR DESCRIPTION
**Pull request by ununhexium**
```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [ ] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [ ] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [ ] I have updated the Community Edition [Postman-Collection](https://github.com/sovity/edc-ce/blob/main/docs/api/postman_collection.json) if I changed existing APIs or added new APIs (e.g. for Management-API or API-Wrapper)
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractAgreementPageQuery.java [hunk 1]
```diff
@@ -26,5 +26,5 @@ public class ContractAgreementPageQuery {
     @Schema(
         description = "Optionally filter the resulting contract agreements by their termination status.",
         requiredMode = Schema.RequiredMode.NOT_REQUIRED)
-    private ContractTerminationStatus terminationStatus;
+    private ContractTerminationStatus status;
```
Comment (richardtreier) [inline extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractAgreementPageQuery.java:29]: is this meant to be part of the PR?

File: extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractTerminationInitiator.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ */
+
+package de.sovity.edc.ext.wrapper.api.ui.model;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+
+public enum ContractTerminationInitiator {
```
Comment (richardtreier) [inline extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractTerminationInitiator.java:18]: This is missing the annotation, especially with the important `enumAsRef = true`

File: extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractTerminationInitiator.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    @Schema(description = "The contract termination was initiated by this EDC.")
+    SELF,
+    @Schema(description = "The contract termination was initiated by the counterparty EDC.")
```
Comment (richardtreier) [inline extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractTerminationInitiator.java:21]: this does nothing, here you can prefer java doc, or skip it

File: docs/api/sovity-edc-api-wrapper.yaml [hunk 4]
```diff
@@ -1351,6 +1351,7 @@ components:
     ContractAgreementTerminationInfo:
       required:
       - detail
+      - initiator
```
Comment (richardtreier) [inline docs/api/sovity-edc-api-wrapper.yaml:1354]: since we do not have a fully managed "process" of termination, I would recommend that we do not coin a new term, and simply go by "terminatedBy" or sth like that.

No "terminators", please 😄 

We'll get DMCA'd

**Review by richardtreier — CHANGES_REQUESTED** [review]

File: extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractAgreementPageQuery.java (hunk 1 above)
Comment (ununhexium) [reply extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractAgreementPageQuery.java]: No, I though I fixed it in the review...

File: extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractTerminationInitiator.java (hunk 3 above)
Comment (ununhexium) [reply extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractTerminationInitiator.java]: Removed.

File: extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractTerminationInitiator.java (hunk 2 above)
Comment (ununhexium) [reply extensions/wrapper/wrapper-api/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractTerminationInitiator.java]: Fixed.
I also fixed the enum in the previous API PR that was missing the same option.

File: docs/api/sovity-edc-api-wrapper.yaml (hunk 4 above)
Comment (ununhexium) [reply docs/api/sovity-edc-api-wrapper.yaml]: Replaced by `terminatedBy` then.

**Review by ununhexium — COMMENTED** [review]
Code review done.

File: extensions/sovity-messenger/src/test/java/de/sovity/edc/extension/messenger/impl/SovityMessengerTest.java [hunk 5]
```diff
@@ -18,6 +18,7 @@
 import de.sovity.edc.extension.messenger.dto.UnsupportedMessage;
 import lombok.val;
 import org.eclipse.edc.spi.message.RemoteMessageDispatcherRegistry;
+import org.eclipse.edc.spi.monitor.ConsoleMonitor;
```
Comment (richardtreier) [inline extensions/sovity-messenger/src/test/java/de/sovity/edc/extension/messenger/impl/SovityMessengerTest.java:21]: not part of this PR?

**Review by richardtreier — CHANGES_REQUESTED** [review]
Looks good, only one place which does not belong in the PR

File: extensions/sovity-messenger/src/test/java/de/sovity/edc/extension/messenger/impl/SovityMessengerTest.java (hunk 5 above)
Comment (ununhexium) [reply extensions/sovity-messenger/src/test/java/de/sovity/edc/extension/messenger/impl/SovityMessengerTest.java]: Yes, I was editing another branch.

**Review by richardtreier — APPROVED** [review]
LGTM
